### PR TITLE
Disable x25519 key exchange in fips mode

### LINKED ISSUE #1992: X25519 might not be FIPS compliant for use in key exchange
**Checklist**
* [X] I've searched the issue tracker for similar bugs.

**Describe the bug**
_Prefacing this with the fact that I am not a compliance expert, apologies if this is not an issue._

`rustls` built with `features = ["fips"]` uses `X25519` as a `kx_group` despite `X25519` not being allowed for use with key exchange under FIPS, at least according to these discussions: https://github.com/openssl/openssl/discussions/22054, https://stackoverflow.com/questions/75875136/about-use-of-secp256r1-and-x25519

**To Reproduce**

```rust
fn main() {
    let provider = rustls::crypto::aws_lc_rs::default_provider();

    println!("FIPS enabled: {}", provider.fips());

    println!("Key exchange groups:");
    for group in provider.kx_groups {
        println!("  {:?}: {}", group, group.fips());
    }
}
```

Output:
```
FIPS enabled: true
Key exchange groups:
  X25519: true
  secp256r1: true
  secp384r1: true
```

**Applicable Version(s)**
Tested on `0.23.8`

**Expected behavior**
The provided discussion links lead me to believe that `X25519` should not be included in the list of `kx_groups` when operating under FIPS. 

**Additional context**
We noticed this when using a FIPS rustls client to talk to a non-rustls FIPS server. The server did not like the `X25519` passed in the TLS 1.3 ClientHello. It responded with a HelloRetryRequest recommending `secp256r1`. This extra round trip defeats the purpose of TLS 1.3.

For now we've worked around this by configuring our own CryptoProvider without `X25519` in the list of kx_groups.

### PR DESCRIPTION
**Pull request by ctz**
This was an oversight during the FIPS work. This PR enables the bogo FIPS compliance tests, which act as a regression test for this change:

```
FAILED (Compliance-fips-202205-TLS-Client-X25519)
unexpected success: local error 'none', child error 'none', stdout:
(...)
FAILED (Compliance-fips-202205-TLS-Server-X25519)
unexpected success: local error 'none', child error 'none', stdout:
(...)
```

fixes #1992

### CODE REVIEW

**Review by cpu — APPROVED** [review]
Thanks for fixing this up. I meant to follow up after the HPKE suite x25519 FIPS compatibility discussion and forgot!

**Comment by cpu** [comment]
This feels worth backporting to supported releases, agree? If it would be helpful I could run with that. Just let me know whether you prefer to do it yourself.

**Comment by ctz** [comment]
> This feels worth backporting to supported releases, agree?

I think 0.23 is the only release line that is affected by this -- 0.23.0 was the first release supporting FIPS.

**Comment by cpu** [comment]
> I think 0.23 is the only release line that is affected by this 

Ah! Great point. I thought it was in 0.22 :+1:

File: rustls/src/crypto/ring/kx.rs [hunk 1]
```diff
@@ -20,6 +20,9 @@ struct KxGroup {
 
     /// The corresponding ring agreement::Algorithm
     agreement_algorithm: &'static agreement::Algorithm,
+
+    /// Whether the algorithm could theoretically be approved.
+    can_be_fips_approved: bool,
```
[the remark this answers is not in the document]
Comment (cpu) [reply rustls/src/crypto/ring/kx.rs:25]: > This file is shared between ring and aws-lc-rs

That makes sense. Thanks!

[the remark this answers is not in the document]
Comment (djc) [reply rustls/src/crypto/ring/kx.rs:25]: Would be nice to make this explicit in the docstring IMO, I had the same reaction.

**Review by djc — APPROVED** [review]

File: rustls/src/crypto/ring/kx.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (ctz) [reply rustls/src/crypto/ring/kx.rs]: How about https://github.com/rustls/rustls/compare/93d7522e8c926297f20c0d55872b8645a16d6183..5bd7a47bea44bfb1fdeb117a48c5778f1b506525 ?

[the remark this answers is not in the document]
Comment (cpu) [reply rustls/src/crypto/ring/kx.rs]: LGTM

**Pull request merged**